local/resource_local_file.go

```diff
@@ -36,6 +36,20 @@ func resourceLocalFile() *schema.Resource {
 				Required:    true,
 				ForceNew:    true,
 			},
+			"file_permission": {
+				Type:        schema.TypeInt,
```

Comment:
It would be best to type these as TypeString.

TypeInt isn't guaranteed to respect numerical base and can lead to unexpected conversion issues between Terraform versions.